sketch/edit_distance.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "sketch/sketch_base.hpp"
+#include "util/multivec.hpp"
+#include "util/timer.hpp"
+#include "util/utils.hpp"
+
+#include <cassert>
+#include <cmath>
+#include <random>
+
+namespace ts { // ts = Tensor Sketch
+
+template <class seq_type>
+class EditDistance : public SketchBase<const std::vector<seq_type> *, false> {
+  public:
+    explicit EditDistance(const std::string &name = "ED")
+        : SketchBase<const std::vector<seq_type> *, false>(name) {
+        init();
+    }
+
+    void init() {}
+
+    const std::vector<seq_type> *compute(const std::vector<seq_type> &seq) { return &seq; }
+
+    static double dist(const std::vector<seq_type> *a, const std::vector<seq_type> *b) {
```

Comment:
any reason for sending these as pointers rather than const references? We typically use pointers for output params.

Reply:
Yeah, this is slightly hacky: the `EditDistance` class `compute` method just returns a pointer to the input, instead of actually sketching the input. The distance must be computed on the return type of `compute`, so `distance` takes two pointers to input sketches and returns the edit distance.